tests/sentry/api/endpoints/test_organization_details.py

```diff
@@ -807,6 +813,85 @@ def test_org_mapping_already_taken(self):
         OrganizationMapping.objects.create(organization_id=999, slug="taken", region_name="us")
         self.get_error_response(self.organization.slug, slug="taken", status_code=409)
 
+    @with_feature("organizations:api-auth-provider")
+    def test_configure_auth_provider(self):
+        old_config = {"domain": "foo.com"}
+        new_config = {"domain": "bar.com"}
+        provider = "google"
+
+        self.get_success_response(
+            self.organization.slug,
+            method="put",
+            providerKey=provider,
+            providerConfig=old_config,
+        )
+        auth_provider = AuthProvider.objects.get(organization_id=self.organization.id)
+        assert auth_provider.provider == provider
+        assert auth_provider.config == {
+            "domains": ["foo.com"],
+            "version": DATA_VERSION,
+            "sentry-source": "api-organization-details",
+        }
+
+        self.get_success_response(
+            self.organization.slug,
+            method="put",
+            providerKey=provider,
+            providerConfig=new_config,
+        )
+        auth_provider = AuthProvider.objects.get(organization_id=self.organization.id)
+        assert auth_provider.provider == provider
+        assert auth_provider.config == {
+            "domains": ["bar.com"],
+            "version": DATA_VERSION,
+            "sentry-source": "api-organization-details",
+        }
+
+    def test_invalid_auth_provider_missing_feature_flag(self):
+        original_queryset = AuthProvider.objects.all()
+
+        self.get_error_response(
+            self.organization.slug,
+            method="put",
+            providerKey="google",
+            providerConfig={"domain": "foo.com"},
+            status_code=400,
+        )
+        assert set(AuthProvider.objects.all()) == set(original_queryset)
```

Comment:
I don't think this is checking what you think it checks. If you go into Django shell:

```
>>> AuthProvider.objects.create(provider="google", config={}, flags=1, organization_id=1)
<AuthProvider at 0x11849e160: id=7, organization_id=1, provider='google'>

>>> curr_set = set(AuthProvider.objects.all())

>>> ap = AuthProvider.objects.get(organization_id=1)
>>> ap.provider = "okta"
>>> ap.save()
>>> ap
<AuthProvider at 0x1184b40a0: id=7, organization_id=1, provider='okta'>

>>> next_set = set(AuthProvider.objects.all())

>>> curr_set == next_set
True
```